cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala

```diff
@@ -130,9 +130,16 @@ class CromwellClient(val cromwellUrl: URL,
     simpleRequest[WorkflowOutputs](outputsEndpoint(workflowId, args))
   }
 
-  def labels(workflowId: WorkflowId, headers: List[HttpHeader] = defaultHeaders)
+  def labels(workflowId: WorkflowId,
+             newLabelsOpt: Option[List[Label]] = None,
+             headers: List[HttpHeader] = defaultHeaders)
             (implicit ec: ExecutionContext): FailureResponseOrT[WorkflowLabels] = {
-    simpleRequest[WorkflowLabels](labelsEndpoint(workflowId), headers=headers)
+    newLabelsOpt match {
```

Comment:
Rather than having an optional parameter which branches the function into a totally unrelated code path, I think I would just add a new method called `updateLabels` with a non-optional `newLabels` parameter.

Reply:
I don't have a strong preference here, but semantically, the methods name `labels` suggests that it's universal. But I can extract addition in the different method.

Reply:
Extracted